action/main.py

```diff
@@ -0,0 +1,38 @@
+import os
+import shlex
+import sys
+from pathlib import Path
+from subprocess import PIPE, STDOUT, run
+
+ACTION_PATH = Path(os.environ["GITHUB_ACTION_PATH"])
+ENV_PATH = ACTION_PATH / ".ruff-env"
+ENV_BIN = ENV_PATH / ("Scripts" if sys.platform == "win32" else "bin")
+OPTIONS = os.getenv("INPUT_OPTIONS", default="")
+SRC = os.getenv("INPUT_SRC", default="")
+VERSION = os.getenv("INPUT_VERSION", default="")
+
+run([sys.executable, "-m", "venv", str(ENV_PATH)], check=True)
+
+version_specifier=""
+# TODO: some form of validation for user input VERSION
```

Comment:
given that `packaging` is not in std, you could check that only `[a-zA-Z0-9_-.]` are used

Reply:
This doesn't concern me greatly, but feels a potentially nice to have.  I sorta take issue with a cruder match that might not actually be a version.  Not sure whether worth adding

Reply:
yeah please add an assert here that there are no strange characters

Reply:
I added a regex that should work with the expected pattern of ruff versions.